eslint-plugin-expensify/CONST.js

```diff
@@ -14,5 +14,6 @@ module.exports = {
 
         // eslint-disable-next-line max-len
         NO_MULTIPLE_API_CALLS: 'Do not call API or DeprecatedAPI multiple times in the same method. The API response should return all the necessary data in a single request, and API calls should not be chained together.',
+        NO_API_SIDE_EFFECTS_ACTIONS: 'Avoid actions with side effects. Each action should have only one API call or one action method call.',
```

Comment:
"side effects" feels like code for some more specific thing that we are trying to avoid.

>Each action should have only one API call or one action method call.

So an action can either call one API or another action method? Why?

Reply:
I have struggled a lot with formulating the error message and I am very open to changing that to something clearer which I believe we can come down to after discussing this rule.

Reply:
>Avoid actions with side effects.

I guess I am hung up on what makes it a "side effect". We don't return a promise from the other methods (only `makeRequestWithSideEffects()`) so they can't actually have any unless they use that method.

>Each action should have only one API call or one action method call.

This seems like the more important thing, but feels like it should also explain a bit more about why we enforce this.